### PR TITLE
Add default remark plugin definition to theme

### PR DESCRIPTION
**Pull request by rogermparent**
Extracted from #3133

This PR adds a default remark transformer definition to our Gatsby theme. It's actually the one from cml.dev and isn't used on dvc.org since dvc.org has some special plugins that cml.dev doesn't, but it seems like the CML version is the one we'd use on most sites anyway.

Ideally, this remark definition can be used on most of our sites that will use this theme.

### CODE REVIEW

File: plugins/gatsby-theme-iterative-docs/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+// noop
```
Comment (rogermparent) [inline plugins/gatsby-theme-iterative-docs/index.js:1]: Just something I noticed in other gatsby plugins/themes that seems to be standard practice if you're not using index.js.

**Review by julieg18 — APPROVED** [review]
Looks good!

**Comment by shcheklein** [comment]
@julieg18 @rogermparent why this is not merged still? :)

**Comment by rogermparent** [comment]
> why this is not merged still?

Same as [#3133](https://github.com/iterative/dvc.org/pull/3133#issuecomment-1019696228), #3170 takes its place.

**Pull request closed without merging**